### ISSUE TITLE
Getting "Error in data[[rowvar]] : attempt to select less than one element in get1index" when using transcript_counts from rnaseq pipeline

### ISSUE THREAD

**Issue by pcantalupo** [nf-core/differentialabundance#140]
### Description of the bug

I posted to Slack about this. https://nfcore.slack.com/archives/C045UNCS5R9/p1687549875802269. 

I ran the rnaseq pipeline to obtain the `salmon.merged.transcript_counts.tsv` file. I renamed it to `matrix.tsv` and selected the first 10,000 rows to create a smaller file for testing and this bug report.  I changed the 1st column name from `tx` to `transcript_id` to match the `--feature_id_col` param (see Nextflow command below) (if I didn't do this I got a different bug that I haven't reported). 

The pipeline failed with this error:
```
Error in data[[rowvar]] :
    attempt to select less than one element in get1index
  Calls: read_delim_flexible -> read.delim -> read.table  
```

I hope I didn't forget anything. Let me know if you need any further information.  Thank you

### Command used and terminal output

```console
nextflow run nf-core/differentialabundance --input $(realpath samplesheet.csv) \
  --contrasts contrasts.csv \
  --matrix matrix.tsv \
  --features_id_col transcript_id \
  --features_type transcript \
  --gtf gencode.v38.primary_assembly.annotation.gtf \
  --outdir results \
  -config htc.config -ansi-log false
```

### Relevant files

Here are all the relevant files

[nextflow.log](https://github.com/nf-core/differentialabundance/files/11874643/nextflow.log)
[contrasts.csv](https://github.com/nf-core/differentialabundance/files/11874771/contrasts.csv)
[samplesheet.csv](https://github.com/nf-core/differentialabundance/files/11874841/samplesheet.csv)
[matrix.csv](https://github.com/nf-core/differentialabundance/files/11874844/matrix.csv) I renamed this to allow upload. Original name is `matrix.tsv`
[htc.config.txt](https://github.com/nf-core/differentialabundance/files/11874846/htc.config.txt) I renamed this to allow upload. Original name is `htc.config`

GTF file can be downloaded here: http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/gencode.v38.primary_assembly.annotation.gtf.gz

### System information

These are the versions used: nextflow 22.10.7 singularity 3.9.6
Hardware: HTC (Linux)
Executor: slurm
Pipeline version: `v1.2.0-g3a849f0`

**Comment by WackerO** [comment]
I investigated and believe this should be solved by the module update in #201; before this, there was a problem with some column names. I will close this issue when the PR is merged; please reopen if the problem continues to appear!

**Comment by WackerO** [comment]
PR merged, closing this :)
